dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -125,22 +126,18 @@ public void populateAllVanityURLsCache() throws DotDataException {
           vanityUrls.stream().map(vanity -> vanity.get("live_inode").toString()).collect(Collectors.toList());
       final List<Contentlet> contentlets = this.contentletAPI.findContentlets(vanityUrlInodes);
 
-      return contentlets.stream().map(contentlet -> new CachedVanityUrl(this.fromContentlet(contentlet))).collect(Collectors.toList());
+      return contentlets.stream().map(contentlet -> new CachedVanityUrl(this.fromContentlet(contentlet))).sorted().collect(Collectors.toList());
 
     } catch (final Exception e) {
       Logger.error(this,
-          String.format("An error occurred when retrieving Vanity URLs: siteId=[%s], " + "languageId=[%s], includeSystemHost=[%s]",
+          String.format("An error occurred when retrieving Vanity URLs: siteId=[%s], languageId=[%s]",
```

Comment:
We should always append `e.getMessage()` to our message so the real root cause does not get eaten higher up.